### ISSUE TITLE
Dummy syntaxes break other packages

### ISSUE THREAD

**Issue by deathaxe** [monokai-pro/sublime-text#178]
see: https://github.com/Thom1729/Sublime-JS-Custom/issues/146

There are other reports about normal syntax definitions not working, because dummy syntaxes of Monokai Pro theme overriding scopes.

**Comment by Monokai** [comment]
Hmm, I remember we had a conversation about this before. All I want is to give icons to certain file names instead of scopes. Currently, to my knowledge, the only way to achieve this is to generate dummy syntaxes. Any other ideas on how to achieve the same without getting in the way of other definitions?

**Comment by deathaxe** [comment]
Auto-generating syntax dummies if no corresponding real syntax definitions exist, is the only viable way.

**Comment by Monokai** [comment]
Would it help if I rename the scope `source.graphql` to `stub.source.graphql` for example?

**Comment by deathaxe** [comment]
Not really. 

It would work around `- include: scope:source.graphql` using dummy syntaxes by accident.

It is however still likely a file is assigned the stub syntax `stub.source.graphql` when opened by user - depending on package and syntax names and loading order.

It would require users to manually re-assign file extensions to the real syntax, which I'd call bad UX.

**Comment by Monokai** [comment]
I agree it's bad UX. And I also think not having icons for certain file names is bad UX. So we're back where we've started:

1. Either have filename-based icons and dummy syntax overrides.
2. Or no syntax overrides, and also no filename-based icons.

I can understand why you're choosing option 2. But I'm getting lots of emails of users asking for filename-based icons. So I'm choosing option 1.

I really wish there was another way. Suggestions welcome.

**Comment by deathaxe** [comment]
Well, there are reasons for A File Icon doing what it does and being designed how it is.

If ST doesn't (and they most likely won't) provide proper infrastructure for "icon themes", a most suitable solution I could think of, was to use A File Icon as middleware which provides infrastructure to handle dummy syntaxes etc. and let themes just provide the icons.

It already prefers themes' icons if they announce icon support via `.supports-a-file-icon-customization` file.

The downside would be only basic icons being displayed without A File Icon being present, while full set of icons being available only with it.

IMHO, there should be only one package/plugin handling dummy syntaxes and file icon preferences in order to avoid conflicts.

**Comment by Monokai** [comment]
I understand your solution, but it's not optimal. There are two things that are holding me back:

1. the user has to install an extra plugin
2. file icons of A file icon (in different colors) appear in between Monokai Pro icons if Monokai Pro hasn't defined them.

It seems there's no solution without any downsides.

**Comment by LimpEmu** [comment]
For me HTML files are broken, I really need this fixed. It's very, very annoying to set the file type manually for one of the most frequently edited file types. For now, I will no longer use Monokai Pro even though I paid for it.
Thank you @deathaxe for letting me know.

**Comment by jowra** [comment]
The HTML-Color-Syntax is broken for me too.

**Comment by olagjo** [comment]
I also finally regained my sanity by uninstalling Monokai Pro now.

I agree with @deathaxe that subtly breaking the behavior of common syntaxes like SQL and HTML in confusing and difficult-to-google ways, introducing the need to re-select the syntax on every file-open is a worse UX than not having icons.

(I suspect that the subtlety of the bug definitely contributes to many more users emailing with "I would like icons" than "My Sublime Text is broken in a confusing way and I can't for the life of me figure out why", skewing the basis of this trade-off)

**Comment by mhenrixon** [comment]
For it was the tailwindcss syntax that stopped working. 

I solved it by using ApplySyntax but honestly, that isn't a good enough solution. 

For reference: https://discord.com/channels/280102180189634562/280157083356233728/1364945487039303702

**Comment by Monokai** [comment]
This functionality has been changed in version 2.1.0. Prior to this version, icons for specific files (e.g. `package.json`, `tailwind.config.mjs`, `requirements.txt`) were enabled by default. From this version onwards, it's an **opt-in** setting. You can add this to your Sublime Text preferences to enable them again:

```json
"monokai_pro_file_based_icons": true
```

These file-based icons rely on special "dummy" syntaxes, which can conflict with other plugins
(like A File Icon). So if you use another icon plugin, it's best to keep this setting off. As a bonus, Monokai Pro is now compatible with A File Icon, which means it uses the dummy syntaxes from that plugin instead of its own and in doing so resolves the conflicts.
